concordium-consensus/src/Concordium/KonsensusV1/Consensus.hs

```diff
@@ -0,0 +1,211 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE TypeFamilies #-}
+
+-- |Consensus V1
+module Concordium.KonsensusV1.Consensus where
+
+import Control.Monad.State
+import Data.Maybe (fromMaybe, isJust)
+
+import qualified Data.Vector as Vector
+import Lens.Micro.Platform
+
+import Concordium.Types
+import Concordium.Types.HashableTo
+import Concordium.Types.Parameters
+import Concordium.Types.Transactions
+import Concordium.Types.Updates (uiHeader, uiPayload, updateType)
+import Concordium.Utils
+
+import Concordium.GlobalState.BlockState
+import qualified Concordium.GlobalState.Persistent.BlockState as PBS
+import Concordium.GlobalState.TransactionTable
+import qualified Concordium.GlobalState.Types as GSTypes
+import Concordium.KonsensusV1.TransactionVerifier
+import Concordium.KonsensusV1.TreeState.Implementation
+import Concordium.KonsensusV1.TreeState.Types
+import Concordium.KonsensusV1.Types
+import Concordium.Scheduler.Types (updateSeqNumber)
+import Concordium.TimeMonad
+import qualified Concordium.TransactionVerification as TVer
+
+-- |Result of attempting to put a
+-- 'BlockItem' into tree state.
+data PutBlockItemResult
+    = -- |The transaction was accepted into
+      -- the tree state.
+      Accepted
+    | -- |The transaction was rejected. See
+      -- the 'TVer.VerificationResult' for why it
+      -- was rejected.
+      Rejected !TVer.VerificationResult
+    | -- |The transaction was already present
+      -- in the tree state.
+      Duplicate
+    | -- |The transaction nonce was obsolete,
+      -- i.e. inferior to the next available nonce.
+      OldNonce
+    deriving (Eq, Show)
+
+-- |Puts a transaction into the pending transaction table
+-- if it's eligible.
+-- A transaction is eligible if the nonce of the transaction is at least
+-- as high as the next available nonce observed with respect to the focus block.
+--
+-- We always check that the nonce of the transaction (if it has origin 'Block') is
+-- at least what is recorded with respect to next available nonce for the
+-- account. This is not necessary for transactions received individually
+-- as they have already been pre-verified, and they will get rejected if they
+-- do not yield exactly the recorded next available nonce.
+--
+-- This is an internal function only and should not be called directly.
+putPendingTransaction ::
```

Comment:
Do not use "put" as a prefix for a function that is not a serializer. (And even then, consider using "serialize" instead). "put" is already overloaded to mean both serialization and setting the contents of a state monad. This is not either of those. Suggest "add" instead.

Reply:
Added "add"